stream_test.go

```diff
@@ -250,8 +250,58 @@ func TestStreamCloseWhileReconnecting(t *testing.T) {
 	}
 }
 
-func mustSubscribe(t *testing.T, url, lastEventId string) *Stream {
-	stream, err := Subscribe(url, lastEventId)
+func TestStreamReadTimeout(t *testing.T) {
+	timeout := time.Millisecond * 200
+	config := StreamConfig{
+		ReadTimeout:  timeout,
+		InitialRetry: time.Millisecond,
+	}
+
+	server := NewServer()
+	server.ReplayAll = true
+	publishedEvent := &publication{data: "123"}
+	repo := NewSliceRepository()
+	repo.Add(eventChannelName, publishedEvent)
+	server.Register(eventChannelName, repo)
+	// this makes it send exactly one event for each connection
+	httpServer := httptest.NewServer(server.Handler(eventChannelName))
+	defer httpServer.Close()
+	defer server.Close()
```

Comment:
This tears things down in a different order than they were created.  Probably doesn't matter but you could put this call right after `NewServer()`.

Reply:
It seems normal to me to tear things down in LIFO order, especially if the second thing created has a dependency on the first thing.

Also note the comments in the previously existing tests from which this setup code was taken: "The server has to be closed before the httpServer is closed. Otherwise the httpServer has still an open connection and it can not close."{

Reply:
Well, I tried changing all of the tests that have this logic to instead do `defer server.Close()` before `defer httpServer.Close()`, and the result is that they all hang. So I'm thinking that the original logic was correct and the comment was backwards.

Reply:
In this case, it looks does look like the server is closed first (it's defer.Close() is called second so it is at the top of the stack), which seems consistent with the comment.  Seems fine if it works.